quint/src/effects/base.ts

```diff
@@ -174,14 +174,39 @@ function bindVariables(name: string, variables: Variables): Either<string, Subst
   }
 }
 
-function variablesNames(variables: Variables): string[] {
+/**
+ * Finds all variable names refered to by some variables
+ *
+ * @param variables the variables to be searched
+ *
+ * @returns a list of names
+ */
+export function variablesNames(variables: Variables): string[] {
```

Comment:
The bodies of `variablesNames` and `stateVariables` look very similar. Would it make sense to generalize these two functions into one, so we can choose between `variables.name` and `variables.vars` in a lambda?